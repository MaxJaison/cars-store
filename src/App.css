header {
  height: 80px;
  padding: 0 24px;
  border-bottom: 1px solid #ededed;
}

.navbar {
  display: flex;
  justify-content: flex-end;
  height: 100%;
  align-items: center;
}

p:not(:last-child) {
  padding-right: 24px;
}

footer {
  height: 80px;
  position: fixed;
  bottom: 0px;
  left: 0px;
  right: 0px;
  margin-bottom: 0px;
  border-top: 1px solid #ededed;
  display: flex;
  justify-content: center;
  align-items: center;
}
