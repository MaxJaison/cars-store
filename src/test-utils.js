import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { applyMiddleware, createStore } from "redux";
import { Provider } from "react-redux";
import reducer from "./reducers";
import thunk from "redux-thunk";
import { BrowserRouter } from "react-router-dom";

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";
export { render };
