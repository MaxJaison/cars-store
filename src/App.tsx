import React from "react";
import "./App.css";
// @ts-ignore
import { Route, Switch, Redirect } from "react-router-dom";
import CarsList from "./Components/CarsList";
import NoMatch from "./Components/NoMatch";
import Car from "./Components/Car";

const App = () => {
  return (
    <>
      <header>
        <nav className="navbar">
          <p>Purchase</p>
          <p>My Orders</p>
          <p>Sell</p>
        </nav>
      </header>
      <Switch>
        <Route exact path="/">
          <Redirect to="/cars" /> :
        </Route>
        <Route exact path="/cars/:id">
          <Car />
        </Route>
        <Route exact path="/cars">
          <CarsList />
        </Route>
        <Route path="*">
          <NoMatch />
        </Route>
      </Switch>
      <footer>
        <p>Copyright 2020</p>
      </footer>
    </>
  );
};

export default App;
