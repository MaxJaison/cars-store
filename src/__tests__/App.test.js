import React from "react";
import { render } from "../test-utils";
import App from "../App";

test("renders App", () => {
  const { container } = render(<App />, {});
  expect(container).toMatchSnapshot();
});
