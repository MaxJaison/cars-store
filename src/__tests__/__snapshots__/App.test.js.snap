// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<div>
  <header>
    <nav
      class="navbar"
    >
      <p>
        Purchase
      </p>
      <p>
        My Orders
      </p>
      <p>
        Sell
      </p>
    </nav>
  </header>
  <main>
    <div
      class="container"
    >
      <div
        class="filters"
      >
        <div
          class="MuiFormControl-root select"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            id="demo-simple-select-outlined-label"
          >
            Color
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-outlined-label"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              role="button"
              tabindex="0"
            >
              <span>
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  Color
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root select"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            id="demo-simple-select-outlined-label"
          >
            Manufacture
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-outlined-label"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              role="button"
              tabindex="0"
            >
              <span>
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  Manufacture
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root select"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink="true"
            id="demo-simple-select-outlined-label"
          >
            Sort By
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-outlined-label demo-simple-select-outlined"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              id="demo-simple-select-outlined"
              role="button"
              tabindex="0"
            >
              None
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="none"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
              >
                <span>
                  Sort By
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-5 MuiButton-text"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Filter
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="cars"
      >
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiCardContent-root content"
          >
            <div
              class="grey-square"
            />
            <div>
              <div
                class="grey-line"
              />
              <div
                class="grey-line"
              />
              <div
                class="grey-line"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
  <footer>
    <p>
      Copyright 2020
    </p>
  </footer>
</div>
`;
